scripts/zones/LaLoff_Amphitheater/mobs/Ark_Angel_GK.lua

```diff
@@ -7,49 +7,58 @@ mixins = { require('scripts/mixins/job_special') }
 ---@type TMobEntity
 local entity = {}
 
--- TODO: Allegedly has a 12 hp/sec regen.  Determine if true, and add to onMobInitialize if so.
--- TODO: Create listener for SCing with other AAs during DM
+local callPetParams =
+{
+    callPetJob = xi.job.DRG,
+}
 
-local function spawnArkAngelPet(mob, target)
+local function spawnArkAngelPet(mob)
     local battlefield = mob:getBattlefield()
-    if battlefield then
-        local battlefieldId   = battlefield:getID()
-        local battlefieldArea = battlefield:getArea()
-        local content         = xi.battlefield.contents[battlefieldId]
-        local selectedPet     = battlefieldId == xi.battlefield.id.DIVINE_MIGHT and 4 or 2
+    if not battlefield then
+        return false
+    end
 
-        mob:setAutoAttackEnabled(false)
-        mob:setMobMod(xi.mobMod.NO_MOVE, 1)
-        mob:injectActionPacket(mob:getID(), 11, 438, 0, 0x18, 101, 732, 55)
+    local battlefieldId   = battlefield:getID()
+    local battlefieldArea = battlefield:getArea()
+    local content         = xi.battlefield.contents[battlefieldId]
 
-        local pet = SpawnMob(content.groups[selectedPet]['mobIds'][battlefieldArea][1])
+    local petGroupIndex   = battlefieldId == xi.battlefield.id.DIVINE_MIGHT and 4 or 2
+    local petId           = content.groups[petGroupIndex]['mobIds'][battlefieldArea][1]
+
+    if xi.mob.callPets(mob, petId, callPetParams) then
+        local pet = GetMobByID(petId)
         if pet then
             battlefield:insertEntity(pet:getTargID(), false, true)
 
             pet:addListener('DEATH', 'AAGK_PET_DEATH', function(petArg)
                 local petBattlefield = petArg:getBattlefield()
-
                 petBattlefield:setLocalVar('petRespawnGK', GetSystemTime() + 30)
             end)
-
-            pet:updateEnmity(target)
         end
 
-        mob:setAutoAttackEnabled(true)
-        mob:setMobMod(xi.mobMod.NO_MOVE, 0)
+        return true
     end
+
+    return false
```

Comment:
This function (and any other copies around) doesnt seem to need to return a value. It isnt stored anywhere.

Remove this return